File: 1338/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(minSetSize([]int{3,3,3,3,5,5,5,2,2,7}))
}

func minSetSize(arr []int) int {
	freq := make(map[int]int)
	for _, num := range arr {
		freq[num]++
	}
	occ := []int{}
	for _, v := range freq {
		occ = append(occ, v)
	}
	sort.Sort(sort.Reverse(sort.IntSlice(occ)))
	cnt, ans := 0, 0
	for _, c := range occ {
		cnt += c
		ans++
		if cnt * 2 >= len(arr) {
			break
		}
	}
	return ans
}

File: 1338/main.rs
use std::collections::HashMap;

struct Solution {}

impl Solution {
    pub fn min_set_size(arr: Vec<i32>) -> i32 {
        let mut freq = HashMap::new();
        for &num in &arr {
            *freq.entry(num).or_insert(0) += 1;
        }
        let mut occ: Vec<i32> = freq.values().cloned().collect();
        occ.sort_by(|a, b| b.cmp(a));
        let mut cnt = 0;
        let mut ans = 0;
        for c in occ {
            cnt += c;
            ans += 1;
            if cnt * 2 >= arr.len() as i32 {
                break;
            }
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::min_set_size(vec![7,7,7,7,7,7]));
}